variants/backend-base/app/views/application/analytics/_head.html.erb

```diff
@@ -0,0 +1,25 @@
+<%# analytics configuration that goes in the <head> %>
+
+<% google_analytics_id = Rails.application.config.google_analytics_id %>
+
+<% if google_analytics_id&.start_with?("GTM-") %>
+  <!-- Google Tag Manager -->
+  <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+      new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+    'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+  })(window,document,'script','dataLayer','<%= google_analytics_id %>');</script>
+  <!-- End Google Tag Manager -->
```

Comment:
I had a quick look at how we implemented this on my most recent client project:

```ruby
  <%= javascript_tag nonce: true, defer: true do -%>
    var dataLayer = [];

    (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
    new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
    'https://www.googletagmanager.com/gtm.js?id='+i+dl;var n=d.querySelector('[nonce]');
    n&&j.setAttribute('nonce',n.nonce||n.getAttribute('nonce'));f.parentNode.insertBefore(j,f);
    })(window,document,'script','dataLayer','GTM-XXXXXX');
  <% end -%>
```
There are two things I think we should steal from this:

1. use `javascript_tag nonce: true, ...` to avoid the CSP yelling at us about an embedded `<script>` without a nonce
2. Define the `dataLayer` variable in JS. GTM does not define this for you.

Reply:
🤔 maybe it would be worth linking to https://github.com/google/data-layer-helper in a comment - it's really the best dev focused docs I've found about how to actually use the dataLayer